### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by ntzwilly**
## Description: 

- Create model classes for all entities as shown in the ERD diagram.
- Set up associations between models.

Create additional methods for: 

  - Users
    - A method that returns the 3 most recent posts for a given user.
  - Posts
    - A method that updates the posts counter for a user.
    - A method that returns the 5 most recent comments for a given post.
  - Comments
    - A method that updates the comments counter for a post.
  - Likes
    - A method that updates the likes counter for a post.
 
Make current user data available to all controllers.

- Create a method `current_user` in the ApplicationController.
- `Current_user` method returns the first user from a database.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
```
Comment (ElijahTobs) [inline app/models/user.rb]: - [x] According to the ERD diagram, the three should have foreign key `author_id`

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  belongs_to :user
+
```
Comment (ElijahTobs) [inline app/models/post.rb]: - [x] This should be:
```ruby
belongs_to :author, class_name: 'User'

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (ElijahTobs) [inline app/models/comment.rb]: - [x] This should be:
```ruby
belongs_to :author, class_name: 'User'

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (ElijahTobs) [inline app/models/like.rb]: - [x] This should be:
```ruby
belongs_to :author, class_name: 'User'

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  def update_posts_counter(value)
+    user.update('posts_counter', value)
+  end
+
```
Comment (ElijahTobs) [inline app/models/post.rb]: - [x] ___[OPTIONAL]___ You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required ⚠️ 🚦
## Hi @ntzwilly  👋

You've done really well so far! 👍 

_However, there are some changes that you are required to make before this project can be approved._

## Required Changes ♻️


_See the inline comments below_


_If anything doesn't seem clear to you, feel free to leave your question or comment in the PR thread below_

#### "Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise."

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
## Happy Coding :computer: 

_Reviewed by_  **Elijah O. Ayandokun**

File: app/models/post.rb (hunk 5 above)
Comment (ntzwilly) [reply app/models/post.rb]: Thank you for the suggestion. It is done

File: app/models/user.rb (hunk 1 above)
Comment (ntzwilly) [reply app/models/user.rb]: Thank you, I added the author_id as a foreign key for the three

File: app/models/post.rb (hunk 2 above)
Comment (ntzwilly) [reply app/models/post.rb]: Done

File: app/models/comment.rb (hunk 3 above)
Comment (ntzwilly) [reply app/models/comment.rb]: Done

File: app/models/like.rb (hunk 4 above)
Comment (ntzwilly) [reply app/models/like.rb]: Done

File: app/models/user.rb (hunk 1 above)
Comment (rloterh) [reply app/models/user.rb]: This issue is well resolved! ✔️

File: app/models/post.rb (hunk 2 above)
Comment (rloterh) [reply app/models/post.rb]: This issue is well resolved! ✔️

File: app/models/comment.rb (hunk 3 above)
Comment (rloterh) [reply app/models/comment.rb]: This issue is well resolved! ✔️

File: app/models/like.rb (hunk 4 above)
Comment (rloterh) [reply app/models/like.rb]: This issue is well resolved! ✔️

File: app/models/post.rb (hunk 5 above)
Comment (rloterh) [reply app/models/post.rb]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Willy, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**